### PR TITLE
i#1569 AArch64: Make dst reg also a src reg for BFM.

### PR DESCRIPTION
**Pull request by fhahn**
The BFM instruction (bitfield move) copies a number of bits from the
source to the destination register, leaving other bits unchanged [1].
This means the destination register must also be a source register,
otherwise DynamoRIO may not restore the original value of the
destination before a BFM instruction.

### CODE REVIEW

File: core/arch/aarch64/encode_gen.h [hunk 1]
```diff
@@ -6445,9 +6499,9 @@ encoder(byte *pc, instr_t *instr)
     case OP_bcond:
         return encode_opnds_bcond(pc, instr, 0x54000000);
     case OP_bfm:
-        if ((enc = encode_opndsgen_13000000(pc, instr, 0x33000000)) != ENCFAIL)
+        if ((enc = encode_opndsgen_33000000(pc, instr, 0x33000000)) != ENCFAIL)
```
Comment (derekbruening) [inline core/arch/aarch64/encode_gen.h]: style violation: "Avoid embedding assignments inside expressions" https://github.com/DynamoRIO/dynamorio/wiki/Code-Style

Comment (fhahn) [inline core/arch/aarch64/encode_gen.h]: This file is autogenerated by codec.py. I'll have to update the script and this will create a big diff.

Comment (egrimley) [inline core/arch/aarch64/encode_gen.h]: That code is automatically generated, of course, so the fix would be in codec.py, which isn't modified by this commit.

Comment (derekbruening) [inline core/arch/aarch64/encode_gen.h]: You mean you'll submit it as a separate PR?  Sure.

**Comment by fhahn** [comment]
I'll put up a patch for the style changes tomorrow.
